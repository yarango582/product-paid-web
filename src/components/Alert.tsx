import React from 'react';
import styles from '../styles/Alert.module.css';

interface AlertProps {
  message: string;
  type: 'success' | 'error';
  onClose: () => void;
}

const Alert: React.FC<AlertProps> = ({ message, type, onClose }) => {
  return (
    <div className={`${styles.alert} ${styles[type]}`}>
      <p className={styles.message}>{message}</p>
      <button className={styles.closeButton} onClick={onClose}>
        &times;
      </button>
    </div>
  );
};

export default Alert;