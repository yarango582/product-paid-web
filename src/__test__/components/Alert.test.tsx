import { render } from "@testing-library/react";
import Alert from "../../components/Alert";

const mockMessage = "Mensaje de prueba";
const mockOnClose = vi.fn();

describe("Alert", () => {
  it("Hace el render correctamente", () => {
    const { asFragment } = render(<Alert message={mockMessage} onClose={mockOnClose} type="success" />);
    expect(asFragment()).toMatchSnapshot();
  })
});