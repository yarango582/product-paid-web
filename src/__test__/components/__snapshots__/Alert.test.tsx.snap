// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert > Hace el render correctamente 1`] = `
<DocumentFragment>
  <div
    class="_alert_e81bc5 _success_e81bc5"
  >
    <p
      class="_message_e81bc5"
    >
      Mensaje de prueba
    </p>
    <button
      class="_closeButton_e81bc5"
    >
      ×
    </button>
  </div>
</DocumentFragment>
`;

exports[`Alert > should render correctly 1`] = `
<div>
  <div
    class="_alert_e81bc5 _success_e81bc5"
  >
    <p
      class="_message_e81bc5"
    >
      Error
    </p>
    <button
      class="_closeButton_e81bc5"
    >
      ×
    </button>
  </div>
</div>
`;
